app/controllers/users_controller.rb

```diff
@@ -7,15 +7,31 @@ def create
       flash[:notice] = "Account Already Exists"
       redirect_to new_user_path
     else
-      user = User.create(user_params)
-      session[:user_id] = user.id
-      redirect_to root_url
+      user = User.new(user_params)
+      if user.valid?
+        user.save
+        session[:user_id] = user.id
+        redirect_to root_path
+      else
+        flash[:notice] = "Please try again."
+        redirect_to new_user_path
+      end
     end
   end
 
   private
 
   def user_params
-    params.require(:signup).permit(:username, :password, :first_name, :last_name, :email)
+    params.require(:signup).permit(:username,
+                                   :password,
+                                   :first_name,
+                                   :last_name,
+                                   :street,
+                                   :city,
+                                   :state,
+                                   :zipcode,
+                                   :country,
+                                   :email
+                                  )
   end
```

Comment:
I like them all in a row. I think it's more readable, plus if theyre all longer, Hound will yell at me.